### PR TITLE
server: add comments for readability.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -99,19 +99,20 @@ func randomBuf(size int) []byte {
 	return buf
 }
 
-func (s *Server) newConn(conn net.Conn) (cc *clientConn, err error) {
+// newConn creates a new *clientConn from a net.Conn.
+// it allocates a connection ID and random salt data for authentication.
```
Comment (zimulala) [inline server/server.go]: s/it/It

File: server/conn.go [hunk 2]
```diff
@@ -480,6 +493,7 @@ func (cc *clientConn) writeReq(filePath string) error {
 	return errors.Trace(cc.flush())
 }
 
+// handleLoadData
```
Comment (zimulala) [inline server/conn.go]: Please complete it.

**Comment by coocood** [comment]
@shenli PTAL

**Comment by zimulala** [comment]
LGTM

File: server/conn.go [hunk 3]
```diff
@@ -480,6 +493,8 @@ func (cc *clientConn) writeReq(filePath string) error {
 	return errors.Trace(cc.flush())
 }
 
+// handleLoadData does the additional work after processed the 'load data' query.
```
Comment (hanfei1991) [inline server/conn.go]: after processing

File: server/conn.go [hunk 4]
```diff
@@ -59,29 +59,34 @@ var defaultCapability = mysql.ClientLongPassword | mysql.ClientLongFlag |
...
-		cc.conn.RemoteAddr(), cc.ctx.Status(), cc.charset, cc.user, cc.ctx.LastInsertID(),
+	collationStr := mysql.Collations[uint8(cc.connectionID)]
+	return fmt.Sprintf("conn: %s, status: %d, collation: %s, user: %s, lastInsertId: %d",
+		cc.conn.RemoteAddr(), cc.ctx.Status(), collationStr, cc.user, cc.ctx.LastInsertID(),
 	)
 }
 
+// handshake works like TCP handshake, but in a higher level, it first writes initial packet to client,
+// during handshake, client and server negotiate compatible features and do authentication.
+// after handshake, client can send sql query to server.
```
Comment (hanfei1991) [inline server/conn.go]: after->After

File: server/conn.go [hunk 5]
```diff
@@ -299,6 +306,9 @@ func (cc *clientConn) readHandshakeResponse() error {
 	return nil
 }
 
+// Run reads client query and writes query result to client in for loop, if there is a panic during query handling,
+// it will be recovered and log the panic error.
+// This function returns and the connection is closed if there is IO error or there is a panic.
```
Comment (hanfei1991) [inline server/conn.go]: an IO error

File: server/conn.go [hunk 6]
```diff
@@ -534,6 +549,9 @@ func (cc *clientConn) handleLoadData(loadDataInfo *executor.LoadDataInfo) error
 	return nil
 }
 
+// handleQuery executes the sql query string and writes result set or result ok to the client.
+// As the execution time of this function represents the performance of TiDB, we do time log and metrics here.
+// There is a special query `load data` that does not returns result, which is handled differently.
```
Comment (hanfei1991) [inline server/conn.go]: returns -> return

File: server/packetio.go [hunk 7]
```diff
@@ -48,6 +48,7 @@ const (
 	defaultWriterSize = 16 * 1024
 )
 
+// packetIO is a helper to read and write data in packet format.
```
Comment (hanfei1991) [inline server/packetio.go]: data -> datas

**Comment by tiancaiamao** [comment]
LGTM

File: server/packetio.go (hunk 7 above)
Comment (coocood) [inline server/packetio.go]: data is not countable.

**Comment by coocood** [comment]
@hanfei1991 PTAL

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**